### PR TITLE
Fix order by price in collection page

### PR DESCRIPTION
**Pull request by NicolasMahe**
### Description

Switch order by `SALES_MIN_UNIT_PRICE_IN_REF` to `BEST_PRICE`.
See https://github.com/liteflow-labs/backend/pull/2947 for explanation about `BEST_PRICE` order by.

### Before

Only sales are displayed first:
Screenshot 2023-09-29 at 21 32 31

DESC filter put asset without offers first:
Screenshot 2023-09-29 at 21 34 01

Random order of asset with auction's bids:
Screenshot 2023-09-29 at 21 33 00

### After
Screenshot 2023-09-29 at 21 32 15

Screenshot 2023-09-29 at 21 33 52

Screenshot 2023-09-29 at 21 33 09

#### Page collection with Foundation collection
- order by creation date (base time): 1.2sec
- old/incorrect order by price time: 2.5sec
- new order by price time: ~~10sec~~ 1sec with recent backend improvements

New order by price is 8x slower than base and 4x slower than old/incorrect order by price.

### Checklist

- [x] Base branch of the PR is `dev`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -60,7 +60,7 @@
     "@graphql-codegen/typescript-react-apollo": "^3.3.7",
     "@graphql-eslint/eslint-plugin": "^3.20.1",
     "@next/bundle-analyzer": "^13.4.9",
-    "@nft/api-graphql": "^1.0.0-beta.34",
+    "@nft/api-graphql": "^1.0.0-beta.41-prerelease-4",
```
Comment (NicolasMahe) [inline package.json:63]: @NicolasMahe do a proper release to avoid using prerelease in starter-kit

File: pages/collection/[chainId]/[id].tsx [hunk 2]
```diff
@@ -91,9 +91,7 @@ const CollectionPage: FC<Props> = ({ now }) => {
   })
 
   const { limit, offset, page } = usePaginateQuery()
-  const orderBy = useOrderByQuery<AssetsOrderBy>(
-    'SALES_MIN_UNIT_PRICE_IN_REF_ASC',
-  )
+  const orderBy = useOrderByQuery<AssetsOrderBy>('BEST_PRICE_ASC')
```
Comment (NicolasMahe) [inline pages/collection/[chainId]/[id].tsx:94]: @antho1404 please confirm the direction of this new order by.
More info on https://github.com/liteflow-labs/backend/pull/2947

Comment (antho1404) [reply pages/collection/[chainId]/[id].tsx:94]: I'm ok with this direction if the question of direction is about having custom sort, no problem with that.

**Comment by NicolasMahe** [comment]
Putting this PR in draft until infra performance related to this order by is optimized

**Comment by NicolasMahe** [comment]
> Putting this PR in draft until infra performance related to this order by is optimized

Backend performance fixed. Merging.

**Pull request merged**